publish.gradle

```diff
@@ -1,132 +1,217 @@
-    ntVersion = project.ntPublishVersion
+def pubVersion
+if (project.hasProperty("publishVersion")) {
+    pubVersion = project.publishVersion
 } else {
-    ntVersion = WPILibVersion.version
+    pubVersion = WPILibVersion.version
 }
 
-if (project.hasProperty("utilPublishVersion")) {
-    utilVersion = project.utilPublishVersion
-} else {
-    utilVersion = "1.0.2${-> getVersion()}"
-}
+def outputsFolder = file("$buildDir/outputs")
 
-def utilFile = file("$buildDir/wpiutil.txt")
-def ntcoreFile = file("$buildDir/ntcore.txt")
+def versionFile = file("$outputsFolder/version.txt")
 
 task outputVersions() {
-    description = 'Prints the versions of ntcore and wpiutil to a file for use by the downstream packaging project'
+    description = 'Prints the versions of wpiutil to a file for use by the downstream packaging project'
     group = 'Build'
-    outputs.files(utilFile, ntcoreFile)
+    outputs.files(versionFile)
 
     doFirst {
-        buildDir.mkdir()
+        outputsFolder.mkdir()
     }
 
     doLast {
-        utilFile.write utilVersion
-        ntcoreFile.write ntVersion
+        versionFile.write pubVersion
     }
 }
 
-task clean(type: Delete) {
-    delete utilFile
-    delete ntcoreFile
-}
+build.dependsOn outputVersions
 
-outputVersions.mustRunAfter clean
+def baseArtifactId = 'ntcore'
+def artifactGroupId = 'edu.wpi.first.ntcore'
 
-project(':native:wpiutil').build.dependsOn outputVersions
-project(':native:ntcore').build.dependsOn outputVersions
-if (project.buildArm) {
-    project(':arm:wpiutil').build.dependsOn outputVersions
-    project(':arm:ntcore').build.dependsOn outputVersions
-}
+def licenseFile = new File("$rootDir/license.txt")
 
+task cppSourcesZip(type: Zip) {
+    destinationDir = outputsFolder
+    classifier = "sources"
 
+    from(licenseFile) {
+        into '/'
+    }
 
-// We change what repo we publish to depending on whether this is a development, beta, stable, or full
-// release. This is set up in the main gradle file.
-publishing {
-    publications {
-        def nat = project('native:ntcore')
-        if (!project.hasProperty('skipJava')) {
-            java(MavenPublication) {
-                artifact nat.jar
-                artifact nat.networktablesJavaSource
-                artifact nat.networktablesJavadoc
-                if (project.buildArm) {
-                    def natArm = project('arm:ntcore')
-                    artifact natArm.jar
-                    // If the library is not embedded include it in the repo
-                    if (!project.hasProperty('compilerPrefix')) {
-                        artifact natArm.ntcoreZip
-                    }
-                }
+    from('src/main/native/cpp') {
+        into '/'
+    }
 
-                if (project.hasProperty('makeDesktop')) {
-                    artifact nat.jar, {
-                        classifier = 'desktop'
+    model {
+        tasks {
+            it.each {
+                if (it.name.contains('jniHeaders')) {
```

Comment:
Reading through that code, just as an aside, its much better practice to create types for your tasks. Doing so allows you to use the annotations `@Input`, `@InputDirectory`, `@Output`, `@OutputDirectory` ect... That allow your build to do things better with task avoidance (speeding up the build).
Additionally, doing so would allow you to take advantage of the new gradle build cache where you can share the output of tasks on a central server (or even on your personal machine) to avoid running tasks that you've already run before.

Additionally, I worry that the lack of any sort of type system here will make this code unmaintainable by anyone in the future. Without a solid set of tests to prove that what the build is creating is correct it's much harder for new developers to dive in without the fear of breaking something.